ext/case_transform/src/lib.rs

```diff
@@ -2,80 +2,128 @@
+use inflector::cases::{camelcase, classcase, kebabcase, snakecase};
+
+use ruru::{Class, Object, VerifiedObject, RString, Hash, Array, Symbol, AnyObject};
 use ruru::types::ValueType;
+use ruru::result::Error as RuruError;
 
-class!(CaseTransform);
+trait Transform: Object {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject;
+}
 
-methods! (
-    CaseTransform,
-    itself,
+impl Transform for AnyObject {
+    fn transform(&self, _transform_function: &Fn(String) -> String) -> AnyObject {
+        self.clone()
+    }
+}
 
-    fn deepTransformKeys(hash: Hash, block: &Fn(String) -> String) -> Hash {
-        let result = Hash::new();
+impl Transform for RString {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject {
+        let result = transform_function(self.to_string();
 
-        hash.unwrap().each(|key, value| {
-            let newValue = if value.ty() == ValueType::Hash { deepTransformKeys(value, block).to_any_object() } else { value };
-            let newKey = RString::new(block(key.unwrap().to_string()));
-            result.store(newKey, newValue);
-        });
+        RString::new(&result)).to_any_object()
+    }
+}
+
+impl Transform for Symbol {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject {
+        let result = transform_function(self.to_string());
 
-        result
+        Symbol::new(&result).to_any_object()
     }
+}
+
+impl Transform for Hash {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject {
+        let mut result = Hash::new();
+
+        self.each(|key, value| {
+            let new_key = transform(key, transform_function);
+            let new_value = match value.ty() {
+                ValueType::Hash => transform(value, transform_function),
+                _ => value,
+            };
 
-    fn transformArray(value: Array, transformMethod: &Fn(AnyObject) -> AnyObject) -> Array {
-        value.map(|item| transformMethod(item)).unwrap()
+            result.store(new_key, new_value);
+        });
+
+        result.to_any_object()
     }
+}
+
+impl Transform for Array {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject {
+        // Temp hack to consume &self for iterator
+        let result = unsafe { self.to_any_object().to::<Array>() };
 
-    fn transformHash(value: Hash, transformMethod: &Fn(AnyObject) -> AnyObject) -> Hash {
-        deepTransformKeys(value, |key| transformMethod(key))
+        result.into_iter()
+            .map(|item| transform(item, transform_function))
+            .collect::<Array>()
+            .to_any_object()
     }
+}
 
-    fn transformSymbol(value: Symbol, transformMethod: &Fn(AnyObject) -> AnyObject) -> Symbol {
-        let transformed = transformMethod(value);
-        Symbol::new(transformed);
+trait TryTransform: Object {
+    fn try_transform<T>(&self,
```

Comment:
`try_transform<T>` method:
 1. Tries to convert current `AnyObject` to the type `T`.
 2. :
    - If the conversion is successful it performs a transformation.
    - otherwise, returns an error from `try_convert_to`.

An example of usage of `try_transform` is

```rust
object.try_transform::<Hash>(key_transform) // if the object is a hash, try to do a hash transformation
```

The interesting part here is the constraints for `T` type: `where T: VerifiedObject + Transform`.